src/spoke/VaultSpoke.sol

```diff
@@ -0,0 +1,434 @@
+    bytes calldata signature
+  ) external returns (uint256) {
+    require(block.timestamp <= params.deadline, InvalidSignature());
+    bytes32 digest = _hashTypedData(params.hash());
+    require(
+      SignatureChecker.isValidSignatureNow(params.owner, digest, signature),
+      InvalidSignature()
+    );
+    _useCheckedNonce(params.owner, params.nonce);
+    return
+      _executeRedeem({
+        caller: params.owner,
+        receiver: params.receiver,
+        owner: params.owner,
+        shares: params.shares
+      });
+  }
+
+  /// @inheritdoc IVaultSpoke
+  function depositWithPermit(
+    uint256 assets,
+    address receiver,
+    uint256 deadline,
+    uint8 v,
+    bytes32 r,
+    bytes32 s
+  ) external returns (uint256) {
+    try
+      IERC20Permit(ASSET).permit({
+        owner: msg.sender, // deposit only mints for caller
+        spender: address(this),
+        value: assets,
+        deadline: deadline,
+        v: v,
+        r: r,
+        s: s
+      })
+    {} catch {}
+    return deposit(assets, receiver);
+  }
+
+  /// @inheritdoc IERC20Permit
+  function permit(
+    address owner,
+    address spender,
+    uint256 value,
+    uint256 deadline,
+    uint8 v,
+    bytes32 r,
+    bytes32 s
+  ) external {
+    require(block.timestamp <= deadline, InvalidSignature());
+    bytes32 digest = _hashTypedData(
+      keccak256(
+        abi.encode(
+          EIP712Hash.PERMIT_TYPEHASH,
+          owner,
+          spender,
+          value,
+          _useNonce({owner: owner, key: PERMIT_NONCE_KEY}),
+          deadline
+        )
+      )
+    );
+    require(owner == ECDSA.recover({hash: digest, v: v, r: r, s: s}), InvalidSignature());
+    _approve({owner: owner, spender: spender, value: value});
+  }
+
+  /// @inheritdoc IVaultSpoke
+  function renounceAllowance(address owner) external override {
+    if (allowance({owner: owner, spender: msg.sender}) == 0) {
+      return;
+    }
+    _approve({owner: owner, spender: msg.sender, value: 0});
+  }
+
+  /// @inheritdoc IERC4626
+  function previewDeposit(uint256 assets) public view virtual returns (uint256) {
+    return HUB.previewAddByAssets(ASSET_ID, assets);
+  }
+
+  /// @inheritdoc IERC4626
+  function previewMint(uint256 shares) public view virtual returns (uint256) {
+    return HUB.previewAddByShares(ASSET_ID, shares);
+  }
+
+  /// @inheritdoc IERC4626
+  function previewWithdraw(uint256 assets) public view virtual returns (uint256) {
+    return HUB.previewRemoveByAssets(ASSET_ID, assets);
+  }
+
+  /// @inheritdoc IERC4626
+  function previewRedeem(uint256 shares) public view virtual returns (uint256) {
+    return HUB.previewRemoveByShares(ASSET_ID, shares);
+  }
+
+  /// @inheritdoc IERC4626
+  function convertToShares(uint256 assets) external view returns (uint256) {
+    return previewDeposit(assets);
+  }
+
+  /// @inheritdoc IERC4626
+  function convertToAssets(uint256 shares) external view returns (uint256) {
+    return previewMint(shares);
```

Comment:
I think this should be `previewRedeem` instead

from the spec

`The amount of assets that the Vault would exchange for the amount of shares provided, in an ideal scenario where all the conditions are met.`

`MUST round down towards 0.`